synapse/replication/resource.py

```diff
@@ -390,7 +390,7 @@ def caches(self, writer, current_token, limit, request_streams):
 
         caches = request_streams.get("caches")
 
-        if caches is not None:
+        if caches is not None and current_position != caches:
```

Comment:
Note that the other streams use a short-circuit inside the storage method to do this.